scripts/backup-restore.sh

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "$CK8S_CONFIG_PATH" ]; then
+  log "ERROR: CK8S_CONFIG_PATH is not set."
+  log "Please set and export the variable before running the script."
+  exit 1
+fi
+
+usage() {
+  echo "Usage: $0 <action> <target>"
+  echo "Actions: backup | restore"
+  echo "Targets: opensearch | harbor | rclone | velero"
+  exit 1
+}
+
+# Sets a timestamp log before every message, also colors it purple to make it easier to distinct from rest of the text because why not
+PURPLE='\033[0;35m'
+NOCOLOR='\033[0m'
+log() {
+  echo -e "[${PURPLE}$(date +'%Y-%m-%d %H:%M:%S')${NOCOLOR}]: $1"
+}
+
+# allows the script to be run from anywhere
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+REPO_ROOT=$(realpath "$SCRIPT_DIR/..")
+CK8S_CMD="$REPO_ROOT/bin/ck8s"
+
+if [ "$#" -ne 2 ]; then
+  usage
+fi
+
+ACTION="$1"
+TARGET="$2"
+
+run_backup() {
+  case "$1" in
+  opensearch)
+    log "Starting OpenSearch backup..."
+
+    local user="admin"
+    local password
+    password=$(sops -d "${CK8S_CONFIG_PATH}/secrets.yaml" | yq '.opensearch.adminPassword')
+    local os_url
+    os_url="https://opensearch.$(yq '.global.opsDomain' "${CK8S_CONFIG_PATH}/common-config.yaml")"
+    log "OpenSearch URL: ${os_url}"
+
+    log "Discovering snapshot repository name..."
+    local snapshot_repo
+    snapshot_repo=$(curl -kL -s -u "${user}:${password}" "${os_url}/_cat/repositories?v" | awk 'NR==2 {print $1}')
+    if [ -z "$snapshot_repo" ]; then
+      log "ERROR: Could not find OpenSearch snapshot repository."
+      exit 1
+    fi
+    log "Found snapshot repository: ${snapshot_repo}"
+
+    local snapshot_name
+    snapshot_name="manual-snapshot-$(date +%Y%m%d%H%M%S)"
```

Comment:
Suggestion: Do we want to enforce UTC to remove any timezone issues?

```suggestion
    snapshot_name="manual-snapshot-$(date --utc +%Y%m%d%H%M%S)Z"
```

Same further down